registry/extension/oras/artifactservice.go

```diff
@@ -72,7 +87,16 @@ func (h *referrersHandler) Referrers(ctx context.Context, revision digest.Digest
 		return nil, err
 	}
 
-	return referrers, nil
+	// sort the list of descriptors that contain the created annotation
+	sort.Slice(referrersSorted, func(i, j int) bool {
+		firstElem, _ := time.Parse(time.RFC3339, referrersSorted[i].Annotations[createAnnotationName])
+		secondElem, _ := time.Parse(time.RFC3339, referrersSorted[j].Annotations[createAnnotationName])
+		// most recent artifact first
+		return firstElem.After(secondElem)
+	})
+	// append the descriptors, which don't have a created annotation, to the end
+	referrersSorted = append(referrersSorted, referrersUnsorted...)
```

Comment:
How do unsorted descriptors work with nextToken? Do you want to sort these by digest (for example) for more deterministic response?

Reply:
Right now the unsorted descriptors are just appended to the end of the referrers list as they are seen when the link files are walked in the filesystem. I can add explicit digest sorting however I think this already done by the storage driver. Since the link files are named by the digest, the walk function when walking the files, first sorts all the child link files lexicographically [here](https://github.com/akashsinghal/distribution/blob/rc1/registry/storage/driver/walk.go#L34).